### ISSUE TITLE
Handle/warn against bad app names

### ISSUE THREAD

**Issue by bunnymatic** [carbonfive/razor#35]
Razor happily created the app for me with the command
```bash
razor new bunnymatic-com
```
in the right directory `bunnymatic-com`, but the files that it writes were unusable
```elixir
# mix.exs
defmodule Bunnymatic-com.Mixfile do
  use Mix.Project

  def project do
    [app: :___b___u___n___n___y___m___a___t___i___c___-___c___o___m___,
     version: "0.0.1",
     elixir: "~> 1.5.1",
     ...
  end
  ...
end
```

```bash
bunnymatic-com (master) $ mix deps.get
warning: variable "com" does not exist and is being expanded to "com()", please use parentheses to remove the ambiguity or change the variable name
  mix.exs:1

** (CompileError) mix.exs:1: undefined function com/0
    (stdlib) lists.erl:1354: :lists.mapfoldl/3
```
Seems like we should warn folks about bad app names or handle them more gracefully.

Is it true that a "valid" name would be camel-case that starts with a capital letter and no punctuation?

Do you have a preference to the "warn" vs "handle" approach?  I might see if I can make up a PR.

**Comment by craiglyons** [comment]
@bunnymatic This is a side effect of a very ungraceful erlang version problem.
With erlang 20, one of Razor's dependencies implodes in this way.

The short-term solution is to downgrade to erlang 19.x. 
For now I note that in `.tool-versions` & readme, but that's just an inadequate bandaid until the dependency is upgraded or avoided.

If you're interested in looking into it, it's called `inflex`. It was bumped a minor version this month, so maybe it's fixed.

**Comment by bunnymatic** [comment]
Noted.  There is a PR for a more like rails inflector with a little progress from about a month ago (https://github.com/nurugger07/inflex/pull/47), but that doesn't seem to be included in the new Inflect bump.  i might still submit what i've got... with hints from that PR on Inflect

**Comment by bunnymatic** [comment]
interestingly, even if i try to match Rails inflector, things don't *quite* do what I expect.  Though the `Inflect` bump might help, i think we either need more cleansing logic around the input name to app name vs dir name, or something else.  

The PR i submitted is not ready for merge but it adds tests that basically match what `ActiveSupport::Inflector` does.  It's a starting point.  I might give this a few more cycles to try to prove my case and show you what I expected...

**Comment by craiglyons** [comment]
This should be fixed in version 0.3.0.

**Comment by bunnymatic** [comment]
It is.  Thanks.
